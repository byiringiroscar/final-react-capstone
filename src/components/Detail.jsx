import React from 'react';
import { useSelector } from 'react-redux';
import { useParams } from 'react-router-dom';
import './detail.css';
import loading from '../images/loading.gif';

const Detail = () => {
  const { games, isLoading } = useSelector((state) => state.games);
  const { id } = useParams();
  const gameDetail = games.find((item) => item.id === parseInt(id, 10)) ?? {};
  if (isLoading) {
    return (
      <main>
        <img className="loading-screen" src={loading} alt="loading" />
      </main>
    );
  }
  return (
    <main className="main-detail">
      <div className="card-container-detail">
        <h2 className="movie-title">{gameDetail.title}</h2>
        <div className="card-detail">
          <div className="card-image-detail">
            <img className="image-detail" src={gameDetail.thumbnail} alt="gameImage" />
            <a target="_blank" href={gameDetail.game_url} rel="noreferrer">Play Game</a>
          </div>
          <div className="card-description">
            <p className="game-description">{gameDetail.short_description}</p>
            <h2 className="additional">Addition Information</h2>
            <div className="info">
              <p className="add">Developer</p>
              <p>{gameDetail.developer}</p>
            </div>
            <div className="info">
              <p className="add">Genre</p>
              <p>{gameDetail.genre}</p>
            </div>
            <div className="info">
              <p className="add">Platform</p>
              <p>{gameDetail.platform}</p>
            </div>
            <div className="info">
              <p className="add">Publisher</p>
              <p>{gameDetail.publisher}</p>
            </div>
            <div className="info">
              <p className="add">Release Date</p>
              <p>{gameDetail.release_date}</p>
            </div>
          </div>
        </div>
      </div>
      <a className="back" href="/">Back</a>
    </main>
  );
};

export default Detail;
