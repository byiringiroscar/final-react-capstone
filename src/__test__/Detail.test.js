import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { store } from '../redux/store';
import Detail from '../components/Detail';

describe('Detail', () => {
  it('Render Detail correctly', () => {
    const detailsComponent = renderer
      .create(
        <BrowserRouter>
          <Provider store={store}>
            <Detail />
          </Provider>
        </BrowserRouter>,
      )
      .toJSON();
    expect(detailsComponent).toMatchSnapshot();
  });
});
