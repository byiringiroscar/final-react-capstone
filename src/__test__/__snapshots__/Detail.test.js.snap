// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail Render Detail correctly 1`] = `
<main
  className="main-detail"
>
  <div
    className="card-container-detail"
  >
    <h2
      className="movie-title"
    />
    <div
      className="card-detail"
    >
      <div
        className="card-image-detail"
      >
        <img
          alt="gameImage"
          className="image-detail"
        />
        <a
          rel="noreferrer"
          target="_blank"
        >
          Play Game
        </a>
      </div>
      <div
        className="card-description"
      >
        <p
          className="game-description"
        />
        <h2
          className="additional"
        >
          Addition Information
        </h2>
        <div
          className="info"
        >
          <p
            className="add"
          >
            Developer
          </p>
          <p />
        </div>
        <div
          className="info"
        >
          <p
            className="add"
          >
            Genre
          </p>
          <p />
        </div>
        <div
          className="info"
        >
          <p
            className="add"
          >
            Platform
          </p>
          <p />
        </div>
        <div
          className="info"
        >
          <p
            className="add"
          >
            Publisher
          </p>
          <p />
        </div>
        <div
          className="info"
        >
          <p
            className="add"
          >
            Release Date
          </p>
          <p />
        </div>
      </div>
    </div>
  </div>
  <a
    className="back"
    href="/"
  >
    Back
  </a>
</main>
`;
